lightly/models/swav.py

```diff
@@ -0,0 +1,55 @@
+import torch
+from torch import nn
+
+from lightly.models.modules import SwaVProjectionHead, SwaVPrototypes
+from lightly.loss.memory_bank import MemoryBankModule
+
+
+class SwaV(nn.Module):
+    def __init__(self, backbone, num_ftrs, out_dim, n_prototypes, queue_length=0):
+        super().__init__()
+        self.backbone = backbone
+        self.projection_head = SwaVProjectionHead(num_ftrs, num_ftrs, out_dim)
+        self.prototypes = SwaVPrototypes(out_dim, n_prototypes=n_prototypes)
+        # Queues are initialized in the first forward call
+        self.queues = []
```

Comment:
I think you could register the queues in the beginning as a [ModuleList](https://pytorch.org/docs/stable/generated/torch.nn.ModuleList.html):

```
self.queues = ModuleList(modules=[MemoryBankModule(size=queue_length) for _ in range(n_queues)])
```

This would make sure that queues are moved to the correct device on `model.to(device)` calls. The drawback is that `n_queues` has to be passed as argument but I think that it is ok to assume that the number of queues is known from the beginning.

What do you think?